### PR TITLE
Feat: Added Currency Converter page

### PR DESCRIPTION
**Pull request by Mayor-Rhymes**
@ArinzeJeffrey-droid. Good evening man. I just added the currency converter page. Can you peruse my code?

### CODE REVIEW

File: src/app/currency-converter/page.tsx [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+    return (
+        <div>
+            <h3 className="text-center text-xl">This is the currency converter section</h3>
+
+
+            <Converter />
+        </div>
+    )
+}
```
Comment (ArinzeJeffrey-droid) [inline src/app/currency-converter/page.tsx]: Indentation could be done better.

File: src/app/note/page.tsx [hunk 2]
```diff
@@ -17,7 +17,7 @@ export default async function Note() {
 
   return (
     <div className="flex flex-col space-y-6">
-      <p className="text-center text-2xl">This is the note section</p>
+      <h3 className="text-center text-xl">This is the note section</h3>
```
Comment (ArinzeJeffrey-droid) [inline src/app/note/page.tsx:20]: Nice!
Using semantics elements for the right content 👏🏾

File: src/components/converter.tsx [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+    const request = await fetch(`${url}`, {
+      headers: {
+        "Content-Type": "application/json",
+        Authorization: `Basic ${encodedCredentials}`,
+      },
+    });
+
+    const data = await request.json();
+    return data;
+  };
```
Comment (ArinzeJeffrey-droid) [inline src/components/converter.tsx]: Move the `fetcher` function outside of the component.
```ts
const fetcher = async (url) => {
  const request = await fetch(`${url}`, {
    headers: {
      "Content-Type": "application/json",
      Authorization: `Basic ${encodedCredentials}`,
    },
  });

  const data = await request.json();
  return data;
};

export default function Converter() {
  // ... rest of the component code ...
}
```

File: src/components/converter.tsx [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+          `https://xecdapi.xe.com/v1/convert_from/?from=${from}&to=${to}&amount=${amount}`
+        );
+
+        if (result) {
+          setResult(result.to[0].mid);
+        } else {
+          console.log("Error");
+        }
+      } catch (err) {
+        console.log("Error");
```
Comment (ArinzeJeffrey-droid) [inline src/components/converter.tsx]: Better error handling rather than console logging?

File: src/components/converter.tsx [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+                await handleConversion();
+                
+              }
+
+              setAmount(event.target.value);
+            }}
+            type="number"
+            name="amount"
+            className="h-[50px] outline-none border-slate-300 rounded-md border-2 p-3"
+          />
```
Comment (ArinzeJeffrey-droid) [inline src/components/converter.tsx]: Maybe add a validation for the amount so the user doesn't enter a negative number?

File: src/app/currency-converter/page.tsx (hunk 1 above)
Comment (Mayor-Rhymes) [reply src/app/currency-converter/page.tsx]: @ArinzeJeffrey-droid. I forgot to format it. My bad. Good morning.

File: src/components/converter.tsx (hunk 4 above)
Comment (Mayor-Rhymes) [reply src/components/converter.tsx]: @ArinzeJeffrey-droid. I am thinking of adding toast notifications to it.

File: src/components/converter.tsx (hunk 5 above)
Comment (Mayor-Rhymes) [reply src/components/converter.tsx]: That is true.

**Comment by Mayor-Rhymes** [comment]
@ArinzeJeffrey-droid. I have made some changes to the code. Thanks for your input. Also, is there a reason why you prefer the fetcher function outside the component? Is it for reusability purpose?

**Comment by ArinzeJeffrey-droid** [comment]
> @ArinzeJeffrey-droid. I have made some changes to the code. Thanks for your input. Also, is there a reason why you prefer the fetcher function outside the component? Is it for reusability purpose?

@Mayor-Rhymes 
To prevent it from being rerendered when state changes in the component, and besides it's just return a data whose value that is not dependent on anything inside of the component block. And also for reusability purpose too.

File: src/components/navbar.tsx [hunk 6]
```diff
@@ -12,11 +12,10 @@ export default function Navbar() {
                 <li><Link href="/" className="hover:bg-slate-300 p-2 hover:rounded-sm transition-all">Home</Link></li>
                 <li><Link href="note" className="hover:bg-slate-300 p-2 hover:rounded-sm transition-all">Note</Link></li>
                 <li><Link href="calculator" className="hover:bg-slate-300 p-2 hover:rounded-sm transition-all">Calculator</Link></li>
-                <li><Link href="" className="hover:bg-slate-300 p-2 hover:rounded-sm transition-all">Switch</Link></li>
-                
+                <li><Link href="/currency-converter" className="hover:bg-slate-300 p-2 hover:rounded-sm transition-all">Currency-Converter</Link></li>
                 <li><Link href="" className="hover:bg-slate-300 p-2 hover:rounded-sm transition-all">Signup/Login</Link></li>
```
Comment (ArinzeJeffrey-droid) [inline src/components/navbar.tsx]: These could probably need some formatting as well too.

**Review by ArinzeJeffrey-droid — APPROVED** [review]

File: src/components/navbar.tsx (hunk 6 above)
Comment (Mayor-Rhymes) [reply src/components/navbar.tsx]: Ok boss.

**Comment by Mayor-Rhymes** [comment]
> > @ArinzeJeffrey-droid. I have made some changes to the code. Thanks for your input. Also, is there a reason why you prefer the fetcher function outside the component? Is it for reusability purpose?
> 
> @Mayor-Rhymes To prevent it from being rerendered when state changes in the component, and besides it's just return a data whose value that is not dependent on anything inside of the component block. And also for reusability purpose too.

@ArinzeJeffrey-droid. Hmmm, I honestly didn't know functions are also rendered. I thought it was only the jsx that is returned from the functions that are rendered. While I am ignorant of what happens at the lower levels in React, rendering functions that are dependent on events just feels wrong.

**Pull request merged**